### ISSUE TITLE
Column Weight problem

### ISSUE THREAD

**Issue by ulaslaslas** [ISchwarz23/SortableTableView#61]
Hi. Thanks for lib.

I using table in horizontal scroolview, on landscape screen. Because i have 9 columns.

But, i can not give it the width I want.

`setColumnWeight(0, 3);
        setColumnWeight(1, 3);
        setColumnWeight(2, 3);
        setColumnWeight(3, 4);
        setColumnWeight(4, 4);
        setColumnWeight(5, 4);
        setColumnWeight(6, 4);
        setColumnWeight(7, 5);
        setColumnWeight(8, 5);
`

What do I have to do to make sure all the columns are visible?

![screenshot_1479380963](https://cloud.githubusercontent.com/assets/12948351/20387394/7ef78830-acc8-11e6-9063-a817b11c28f6.png)
![screenshot_1479380968](https://cloud.githubusercontent.com/assets/12948351/20387395/7efa3b02-acc8-11e6-8b84-675d0d5555c5.png)

**Comment by ISchwarz23** [comment]
Hi @ulaslaslas,

please update to the version 2.4.0 of this library and use the `TableColumnWidthModel`. This `TableColumnModel` implementation allows to define absolute widths for each column.  

Usage is simple as that:
```java
TableColumnWidthModel tableColumnModel = new TableColumnWidthModel( 9 );
tableColumnModel.setColumnWidth( 0, 300 );
// define other column widths here 
setColumnModel( tableColumnModel );
```

Please tell me if this fulfilled your needs.

Best regards,
Ingo

**Comment by ulaslaslas** [comment]
Yes!!! @ISchwarz23 
Thanks again.
![screenshot_1479456748](https://cloud.githubusercontent.com/assets/12948351/20423366/827fe820-ad78-11e6-9a9e-41a481917a68.png)
![screenshot_1479456757](https://cloud.githubusercontent.com/assets/12948351/20423367/82805aa8-ad78-11e6-949b-52ba36c6eebf.png)

**Comment by ulaslaslas** [comment]
@ISchwarz23 but :/

There was a problem with the screens in different resolutions.
![23232](https://cloud.githubusercontent.com/assets/12948351/20423975/2102776c-ad7c-11e6-9ef9-d50d80bff8b5.PNG)

**Comment by ISchwarz23** [comment]
I'm currently working on a soltion. A version 2.4.1 will be released today or latest tomorrow :)

**Comment by ISchwarz23** [comment]
Hi @ulaslaslas,

please update to the version 2.4.1 of this library and use the `TableColumnDpWidthModel`. This `TableColumnModel` implementation allows to define absolute widths for each column in density-independent pixels.

Usage is simple as that:

```java
TableColumnDpWidthModel tableColumnModel = new TableColumnDpWidthModel( getContext(), 9 );
tableColumnModel.setColumnWidth( 0, 200 );
// define other column widths here 
setColumnModel( tableColumnModel );
```
Please tell me if this fulfilled your needs **now** completely :P

Best regards,
Ingo

**Comment by ulaslaslas** [comment]
Yes!!! You are the best. 
Certainly one of the most useful libraries.

Problem resolved.

Let's code

![image](https://cloud.githubusercontent.com/assets/12948351/20472458/a9d795ac-afc2-11e6-94b2-f934b705c09a.png)

**Comment by ISchwarz23** [comment]
Thank you very much :)

**Comment by ulaslaslas** [comment]
Hi again. 
İ get errors on different screens. 
Tablecolumnwidthdp method is give dp. But inside Convert to pixel. Why? 
Andorid Layout_width needed dp. But columns change dp to px 😕 Screenshot from asus zenfone2
![img_2110](https://cloud.githubusercontent.com/assets/12948351/20535357/19a25c0c-b0ed-11e6-88d5-fbba23c615e6.JPG)
![img_2114](https://cloud.githubusercontent.com/assets/12948351/20535358/19d7c252-b0ed-11e6-9fd3-aa4da5ff0377.PNG)

**Comment by ISchwarz23** [comment]
Yes, internally the `TableView` is handling all widths in pixels. The different `TableColumnModel` as an abstraction, that is holding the defined column widths in different presentations e.g. px and dp.  

But I didn't understand exactly where the issue is. What have you expected, and what differs from that expectation?

**Comment by ulaslaslas** [comment]
Both are 200dp in width as in the picture. But it is not the same width. The following TextView show my total values. And it needs to be aligned with the column. :/

![19a25c0c-b0ed-11e6-88d5-fbba23c615e6](https://cloud.githubusercontent.com/assets/12948351/20565187/2a7006ea-b199-11e6-8a2f-4513c8bf7c44.JPG)

**Comment by ISchwarz23** [comment]
Hi again :)

execuse me, but I have to ask this question:
Are you sure you are using the TableColumn**Dp**WidthModel and not the TableColumn**Px**WidthModel?  

(I have to ask this before I start the investigation)

**Comment by ulaslaslas** [comment]
Yes. I m using TableColumnDpWidthModel. But you call toPixel(int dp) inside the class 😕

**Comment by ulaslaslas** [comment]
`private int toPixel(final int dp) {
        return Math.round(dp * (displayMetrics.xdpi / DisplayMetrics.DENSITY_DEFAULT));
    }`

I think the problem is here. Because DENSITY_DEFAULT = 160 .  But this value is different for each device.
400 for asus zenfone2 .
